docs/concepts/models/model_kinds.md

```diff
@@ -261,5 +261,273 @@ FROM db.employees;
 ## SEED
 The `SEED` model kind is used to specify [seed models](./seed_models.md) for using static CSV datasets in your SQLMesh project.
 
+## SCD Type 2
+
+SCD Type 2 is a model kind that supports [slowly changing dimensions](https://en.wikipedia.org/wiki/Slowly_changing_dimension#Type_2:_add_new_row) (SCDs) in your SQLMesh project. SCDs are a common pattern in data warehousing that allow you to track changes to records over time. 
+
+SQLMesh achieves this by adding a `valid_from` and `valid_to` column to your model. The `valid_from` column is the date that the record became valid (inclusive) and the `valid_to` column is the date that the record became invalid (exclusive). The `valid_to` column is set to `NULL` for the latest record.
+
+Therefore you can use these models to not only tell you what the latest value is for a given record but also what the values were anytime in the past. Note that maintaining this history does come at a cost of increased storage and compute and this may not be a good fit for sources that change frequently since the history could get very large. 
+
+This example specifies a `SCD_TYPE_2` model kind:
+```sql linenums="1" hl_lines="3"
+MODEL (
+  name db.menu_items,
+  kind SCD_TYPE_2 (
+    unique_key (id),
+  )
+);
+
+SELECT
+  id::INT,
+  name::STRING,
+  price::DOUBLE,
+  updated_at::TIMESTAMP
+FROM 
+  stg.current_menu_items;
+```
+
+SQLMesh will materialize this table with the following structure:
+```sql linenums="1" hl_lines="3"
+TABLE db.menu_items (
+  id INT,
+  name STRING,
+  price DOUBLE,
+  updated_at TIMESTAMP,
```

Comment:
is updated_at mandatory?

Reply:
Yes I assert this when trying to evaluate an SCD Type 2.

Reply:
maybe put that in the initial description